engine: 24.0.6 release notes (#18110)
docker/docs: content/engine/release-notes/24.0.md

- [moby/moby, 24.0.6 milestone](https://github.com/moby/moby/issues?q=is%3Aclosed+milestone%3A24.0.6)

### Bug fixes and enhancements

* containerd storage backend: Fix `docker ps` failing when a container image is no longer present in the content store. [moby/moby#46095](https://github.com/moby/moby/pull/46095)
* containerd storage backend: Fix `docker ps -s -a` and `docker container prune` failing when a container image config is no longer present in the content store. [moby/moby#46097](https://github.com/moby/moby/pull/46097)
* containerd storage backend: Fix `docker inspect` failing when a container image config is no longer (or was never) present in the content store. [moby/moby#46244](https://github.com/moby/moby/pull/46244)
* containerd storage backend: Fix diff and export with the `overlayfs` snapshotter by using reference-counted rootfs mounts. [moby/moby#46266](https://github.com/moby/moby/pull/46266)
* containerd storage backend: Fix a misleading error message when the image platforms available locally do not match the desired platform. [moby/moby#46300](https://github.com/moby/moby/pull/46300)

> * containerd storage backend: Fix building images `FROM scratch` with the classic builder. [moby/moby#46302](https://github.com/moby/moby/pull/46302)

Reviewer: ```suggestion
* containerd storage backend: Fix the `FROM scratch` Dockerfile instruction with the classic builder. [moby/moby#46302](https://github.com/moby/moby/pull/46302)
```

* containerd storage backend: Fix `mismatched image rootfs and manifest layers` errors with the classic builder. [moby/moby#46310](https://github.com/moby/moby/pull/46310)
* Warn when pulling Docker Image Format v1, and Docker Image manifest version 2, schema 1 images from all registries. [moby/moby#46290](https://github.com/moby/moby/pull/46290)
* Fix live-restore of volumes with custom volume options. [moby/moby#46366](https://github.com/moby/moby/pull/46366)
* Fix incorrectly dropping capabilities bits when running a container as a non-root user (note: this change was already effectively present due to a regression). [moby/moby#46221](https://github.com/moby/moby/pull/46221)

> * Fix network isolation iptables rules preventing IPv6 Neighbor Solicitation packets from being exchanged between containers. [moby/moby#46214](https://github.com/moby/moby/pull/46214)

Reviewer: ```suggestion
* Fix network isolation iptables rules preventing IPv6 neighbor solicitation packets from being exchanged between containers. [moby/moby#46214](https://github.com/moby/moby/pull/46214)
```

Author: This is titlecased/a proper noun in the standard, so I believe it should stay as-is: https://datatracker.ietf.org/doc/rfc4861/#:~:text=Neighbor%20Solicitation%3A%20Sent,Duplicate%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20Address%20Detection.

Reviewer: OK; agree

* Fix `dockerd.exe --register-service` not working when the binary is in the current directory on Windows. [moby/moby#46215](https://github.com/moby/moby/pull/46215)
* Add a hint suggesting the use of a PAT to `docker login` against Docker Hub. [docker/cli#4500](https://github.com/docker/cli/pull/4500)

> * Improve shell startup time for users of the Bash completions. [docker/cli#4517](https://github.com/docker/cli/pull/4517)

Reviewer: ```suggestion
* Improve shell startup time for users of Bash completion. [docker/cli#4517](https://github.com/docker/cli/pull/4517)
```

Author: The intention is to convey `for users of Bash completion [for `docker`].`; do you think that your suggestion is better at that? A definite article seems better to me.

Reviewer: I think `the Bash completions` reads strangely. Maybe my suggestion made this too ambiguous. Why not clarify:
```suggestion
* Improve shell startup time for users of Bash completion for the Docker CLI. [docker/cli#4517](https://github.com/docker/cli/pull/4517)
```

Author: That seems fine to me.
